sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -332,6 +339,10 @@ private[hive] class SparkExecuteStatementOperation(
           HiveThriftServer2.eventManager.onStatementFinish(statementId)
         }
       }
+      if (cleanShuffleDeps && rdd != null) {
+        rdd.cleanShuffleDependencies()
+        rdd = null
```

Comment:
When `THRIFTSERVER_INCREMENTAL_COLLECT` is `true`, we will end up cleaning up dependencies before we have used up the data from df.

Reply:
thanks for the input